app/models/notification.rb

```diff
@@ -115,13 +115,13 @@ def remove_all_by_action_without_delay(notifiable_ids:, notifiable_type:, action
     def remove_all(notifiable_ids:, notifiable_type:)
       return unless %w[Article Comment Mention].include?(notifiable_type) && notifiable_ids.present?
 
-      Notifications::RemoveAllJob.perform_later(notifiable_ids, notifiable_type)
+      Notifications::RemoveAllWorker.perform_async(notifiable_ids, notifiable_type)
     end
 
     def remove_all_without_delay(notifiable_ids:, notifiable_type:)
       return unless %w[Article Comment Mention].include?(notifiable_type) && notifiable_ids.present?
 
-      Notifications::RemoveAllJob.perform_now(notifiable_ids, notifiable_type)
+      Notifications::RemoveAll.call(Array.wrap(notifiable_ids), notifiable_type)
```

Comment:
Here I believe you are going to want so that we are still using the same code path as above, we are just executing it immediately. 
```ruby
Notifications::RemoveAll.new.perform(notifiable_ids, notifiable_type)
```
The reason for this is that if we add logic to that job we will want it also used here as well.

Reply:
I didn't get one thing. `RemoveAll` doesn't have `perform` method, how it can be executed?

Reply:
`Notifications::RemoveAllWorker.new.perform(notifiable_ids, notifiable_type)` sorry that was a typo!

Reply:
No problem. I will send the changes.

Reply:
@mstruve I think we gave @luchiago two conflicting reviews :) In my review here - https://github.com/thepracticaldev/dev.to/pull/5451#pullrequestreview-341813877 - I suggested using the service class directly in the model method because I didn't see the point in instantiating the job (the job checks the `notifiable_type` is the correct one and then calls the service, which `remove_all_without_delay` already does).

Either way it's fine, just wanted to point this out :D

Reply:
Oh snap, my bad!!! I do think we want to instantiate the worker in case we decide to add more logic to that worker, I would hate to have it missed here.

Reply:
You raise a valid point, I actually think workers should have only logic related to workers (retry and similar things) and the business logic should be external but we'll get there :)

Reply:
@rhymes @mstruve So, should I keep this last change on my PR or this [#5451 (review)](https://github.com/thepracticaldev/dev.to/pull/5451#pullrequestreview-341813877)?

Reply:
In this case, change it to:
```ruby 
Notifications::RemoveAllWorker.new.perform(notifiable_ids, notifiable_type)
```

Reply:
All right!